cdap-app-fabric/src/main/java/co/cask/cdap/internal/filesystem/LocationCodec.java

```diff
@@ -49,6 +51,11 @@ public JsonElement serialize(Location src, Type typeOfSrc, JsonSerializationCont
   public Location deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) {
     JsonObject jsonObj = json.getAsJsonObject();
     String uri = jsonObj.get("uri").getAsString();
-    return lf.create(uri);
+    try {
```

Comment:
yup the string version is interpreted as a path, so the whole thing doesn't work (this class apparently wasn't used anywhere before this PR). It would deserialize 'file:/my/path' to a location of '/my/path/file:/my/path'.